### PR TITLE
Refactor scripts & tools

### PR DESCRIPTION
**Pull request by anthony-c-martin**
* Use typescript for tests
* Break up test case files
* Restructure project
* Add ajv tests to catch missing/invalid references
* Add express server for local testing

### CODE REVIEW

File: tools/package.json [hunk 1]
```diff
@@ -5,18 +5,13 @@
...
-    "test": "mocha runSchemaValidation.js"
-  },
-  "dependencies": {
-    "chalk": "^1.1.3",
-    "sync-request": "^2.1.0",
-    "tv4": "1.2.7",
-    "vscode-json-languageservice": "3.0.2",
-    "mocha-each": "^1.1.0"
+    "test": "mocha runSchemaValidation.js",
+    "test-ci": "mocha runSchemaValidation.js --reporter mocha-junit-reporter"
```
Comment (Tiano2017) [inline tools/package.json]: >--reporter mocha-junit-reporter [](start = 45, length = 31)

how does this work?

Comment (anthony-c-martin) [reply tools/package.json]: It outputs a file in a standard format (Junit test results) that Azure DevOps knows how to consume

File: tests/2016-03-01/Microsoft.Insights.alertrules.tests.json [hunk 2]
```diff
@@ -1,99 +1,99 @@
...
-              "odata.type": "Microsoft.Azure.Management.Insights.Models.RuleWebhookAction",
-              "serviceUri": "https://localhost/foo"
-            }
-          ]
-        }
-      }
-    }
-
-  ]
+{
```
Comment (Tiano2017) [inline tests/2016-03-01/Microsoft.Insights.alertrules.tests.json]: Is anything changed in this test file?

Comment (anthony-c-martin) [reply tests/2016-03-01/Microsoft.Insights.alertrules.tests.json]: image
Nope. I'm assuming VSCode did some auto-formatting on it.

**Pull request merged**